bindings/rust/wasmedge-macro/src/lib.rs

```diff
@@ -0,0 +1,258 @@
+                fn #outer_fn_name_ident (#outer_fn_inputs) #outer_fn_return {
+                    fn #inner_fn_name_ident (#inner_fn_inputs) #inner_fn_return {
+                        #inner_fn_block
+                    }
+                    #inner_fn_name_ident(#(#args),*)
+                }
+            )
+        }
+    };
+
+    Ok(ret)
+}
+
+#[proc_macro_attribute]
+pub fn async_host_function2(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let body_ast = parse_macro_input!(item as Item);
+    if let Item::Fn(item_fn) = body_ast {
+        match expand_async_host_func2(&item_fn) {
+            Ok(token_stream) => token_stream.into(),
+            Err(err) => err.to_compile_error().into(),
+        }
+    } else {
+        TokenStream::new()
+    }
+}
+
+fn expand_async_host_func2(item_fn: &syn::ItemFn) -> syn::Result<proc_macro2::TokenStream> {
+    let outer_fn_name_ident = &item_fn.sig.ident;
+    let outer_fn_name_literal = outer_fn_name_ident.to_string();
+    let outer_fn_inputs = &item_fn.sig.inputs;
+    let args = outer_fn_inputs.iter().map(|fn_arg| match fn_arg {
+        FnArg::Typed(PatType { pat, .. }) => match &**pat {
+            Pat::Ident(ident) => ident,
+            _ => panic!("argument pattern is not a simple ident"),
+        },
+        FnArg::Receiver(_) => panic!("argument is a receiver"),
+    });
+    let inner_fn_name_literal = format!("inner_{}", outer_fn_name_literal);
+    let inner_fn_name_ident = syn::Ident::new(&inner_fn_name_literal, item_fn.sig.span());
+    let inner_fn_inputs = &item_fn.sig.inputs;
+    let inner_fn_return = &item_fn.sig.output;
+    let inner_fn_block = &item_fn.block;
+
+    let ret = quote!(
+        fn #outer_fn_name_ident (#outer_fn_inputs) -> wasmedge_sys::WasmEdgeHostFuncFuture {
+            async fn #inner_fn_name_ident (#inner_fn_inputs) #inner_fn_return {
+                #inner_fn_block
+            }
+
+            let stack = switcher2::stack::EightMbStack::new().unwrap();
+            AsyncWasmEdgeResult::<switcher2::stack::EightMbStack, wasmedge_sys::WasmEdgeHostFuncResult<Vec<wasmedge_sys::WasmValue>>, fn()>::new(
+                stack,
+                |mut yielder| -> Result<Vec<wasmedge_sys::WasmValue>, wasmedge_types::error::HostFuncError> {
+                    yielder.async_suspend(#inner_fn_name_ident(#(#args),*))
+                },
+            )
+            .unwrap()
+        }
+    );
+
+    Ok(ret)
+}
+
+#[proc_macro_attribute]
+pub fn sys_async_host_function(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    let body_ast = parse_macro_input!(item as Item);
+    if let Item::Fn(item_fn) = body_ast {
+        match sys_async_expand_host_func(&item_fn) {
+            Ok(token_stream) => token_stream.into(),
+            Err(err) => err.to_compile_error().into(),
+        }
+    } else {
+        TokenStream::new()
+    }
+}
+
+fn sys_async_expand_host_func(item_fn: &syn::ItemFn) -> syn::Result<proc_macro2::TokenStream> {
```

Comment:
What's the naming rule?
I found `expand_async_host_func2` and `async_host_func2`, however, these are `sys_async_host_function` and `sys_async_host_function`.

Reply:
The `async_host_func2` macro is defined for declaring async host function when using `sdk`, while `sys_async_host_function` and `sys_async_host_function2` are two different macro types and used for declaring async host function when using `sys`. These three macros are used in another branch. As for `expand_async_host_func2`, it is just a private function used by the `async_host_func2` macro, and not exposed to the outside. Thanks!